packages/ec-core/src/operator/selector/mod.rs

```diff
@@ -13,13 +13,64 @@ pub mod worst;
 
 pub use error::EmptyPopulation;
 
+/// Select an individual from a `Population`.
+///
+/// See [`Select`] for a wrapper that converts a `Selector` into an
+/// [`Operator`], allowing selectors to be used in chains of operators.
+///
+/// # Examples
+///
+/// In this example we use the `[Best]` selector to choose the
+/// "best" (maximal) value from a list.
+///
+/// ```
+/// # use ec_core::operator::selector::{best::Best, Selector};
+/// # use rand::thread_rng;
+/// #
+/// let population = vec![5, 8, 9, 2, 3, 6];
+/// let winner = Best.select(&population, &mut thread_rng()).unwrap();
+/// assert_eq!(*winner, 9);
+/// ```
+///
+/// Here we implement a `First` selector that always returns the first
+/// element in a vector.
+///
+/// ```
+/// # use rand::thread_rng;
+/// # use rand::rngs::ThreadRng;
+/// # use anyhow::Context;
+/// # use ec_core::operator::selector::Selector;
+/// #
+/// struct First;
+///
+/// impl Selector<Vec<u8>> for First {
```

Comment:
Vec?

Reply:
Nope – here again the lack of an implementation of Population for arrays means that we can't replace `Vec` with an array type here.